### ISSUE TITLE
multiple kernels execution cannot be scheduled in parallel using xcl_run_kernel3d() API

### ISSUE THREAD

**Issue by heeranand** [Xilinx/SDAccel_Examples#8]
xcl_run_kernel3d() API in libs/xcl/xcl.c file is a blocking call. As a result, Multiple kernels cannot run in parallel using this API.

**Comment by Spenser309** [comment]
Yes, but this would also require modification to the queue to support out of order queues.  I have a patch for this in my local sandbox and will submit it soon.

**Comment by Spenser309** [comment]
Won't fix. In order to maintain the simplicity of this API non-blocking calls will need to be run using clEnqueueNDRange directly.
